### ISSUE TITLE
Is it possible to use Compose Multiplatform UI as an interface in Godot with this project?

### ISSUE THREAD

**Issue by unicornlox** [utopia-rise/godot-kotlin-jvm#561]
hello, is it possible to use Compose Multiplatform UI as an interface in godot game with this project or how can it be made possible?

**Comment by MartinHaeusler** [comment]
I'm not an expert but: I highly doubt it is possible. Compose Multiplatform UI relies on different UI technologies depending on the platform:

- When used on a desktop (windows, mac, linux), it relies on `java.swing`
- When used in the web, it relies on HTML
- When used on android / iOS it relies on the respective native libraries

Godot brings its own UI system. While I do think that it might be possible (in theory) to write a backend for Compose Multiplatform UI that uses the Godot UI in the background, that would be a huge project in its own right.

Also: I don't think it's a good idea. Godot is a game engine with a main loop that runs once per frame. The UI is therefore also updated once per frame, always. This is contrary to regular UI frameworks (such as Compose) which try to limit re-drawing as much as possible.

Long story short: do yourself a favor and familiarize yourself with the Godot UI instead, it's not all that bad :)

**Comment by unicornlox** [comment]
@MartinHaeusler  godot control node elements look very bad when the screen resolution changes and I have to keep track of the resolution of all nodes with code.I read that haxeui is available in hxgodot and I wondered if it was also possible in kotlin. thanks.

**Comment by chippmann** [comment]
Theoretically this is possible but as @MartinHaeusler already mentioned not really feasible.
For one, you cannot directly interact with godot or the kotlin code which runs in godot from compose and vice versa. They function completely different.

So I highly advice against it.

**Comment by stashymane** [comment]
Although using Compose Multiplatform in its entirety is very unlikely, you can definitely use the Compose compiler to create a Compose API for Godot UI components.
